### ISSUE TITLE
S4: What was the reason/rationale for using S4?

### ISSUE THREAD

**Issue by HenrikBengtsson** [pneuvial/c3co#30]
I probably asked before but I don't remember the answer; what's the reason for using the S4 class system (rather than S3)?  It don't see anything that is making use of S4-unique features.

**Comment by pneuvial** [comment]
Initially there were no classes at all, and Julien suggested to define objects and methods for
plotting/formatting. I don't remember why we chose S4 and not S3.

**Comment by HenrikBengtsson** [comment]
Ok. I agree the use of classes/methods is very useful here.  Disclaimer: I'm not a major fan of S4 (although I was one of the first people in the world waiting for it/and trying it out).

Are there any objections if I go ahead and add S3 class attributes to some of the data structures that are currently just lists (and lists of lists)?  For instance, what's currently returned by `segmentData()` and it's components.

**Comment by jchiquet** [comment]
Yep, totally agree : S4 is a real pain... I personally like R6 (or refClass) much for more OOP formalism, but I am not sure it is relevant for c3co. It probably depends if some objets / structures are common with joinSeg.

Anyway, S3 sounds good to me.

**Comment by pneuvial** [comment]
Currently there is no class/methods in jointSeg, so not much to share yet!

**Comment by pneuvial** [comment]
and thanks @HenrikBengtsson for offering to S3-ify this.
